syntax = "proto3";

package chat;

service ChatService {
  // Stream responses for a query
  rpc ProcessQuery (QueryRequest) returns (stream QueryResponse) {}
}

message QueryRequest {
  string query = 1;
  bool skip_human_verification = 2;
}

message QueryResponse {
  oneof response {
    string text = 1;
    ToolExecution tool = 2;
    HumanVerification verification = 3;
  }
}

message ToolExecution {
  string tool_name = 1;
  string result = 2;
}

message HumanVerification {
  string tool_name = 1;
  map<string, string> arguments = 2;
  string description = 3;
}